core/templates/pages/exploration-editor-page/modal-templates/lost-changes-modal.component.ts

```diff
@@ -44,11 +47,21 @@ export class LostChangesModalComponent
     this.hasLostChanges = (this.lostChanges && this.lostChanges.length > 0);
     this.lostChanges = this.lostChanges.map(
       this.lostChangeObjectFactory.createNew);
-    this.loggerService.error(
-      'Lost changes: ' + JSON.stringify(this.lostChanges));
   }
 
   cancel(): void {
     this.ngbActiveModal.dismiss();
   }
+
+  exportChangesAndClose(): void {
+    let lostChangesData: HTMLElement = (
+      this.elRef.nativeElement.getElementsByClassName(
+        'oppia-lost-changes'));
```

Comment:
Why extracting lost changes from a DOM? Can't we use this.lostChanges?

Reply:
this.lostChanges returns the JSON dictionary but we need to show in the file in the exact same way we are showing in the modal and that's why extracting the template from the DOM.

Reply:
Is it possible to generate a human-readable formataed dict here and use it in json + HTML? [It will be better than manipulating/creating DOM!]

Reply:
I'm okay with the existing format if the frontend team approves the changes from a security and structure perspective.

/cc @srijanreddy98 Can you please check this function and let is know whether it's fine to have it as-is? If not, do you have any suggestions?

Reply:
@srijanreddy98 can you PTAL?

Reply:
For some background: We parse and display the lostChanges object in a particular way in the template. The intention here is to let the user save a copy of what they are seeing in the modal (so that they can use this data later on to manually replicate their changes). We are only reading from the DOM for this so I don't think it is a security issue. The "download" operation in lines 61-64 is a common technique to save data as a file without making a server call. One thing we can do to improve this is a cleanup -- e.g. remove the A tag and revoke the object URL (example: https://gist.github.com/danallison/3ec9d5314788b337b682#file-downloadstring-js-L11-L12). But I don't expect this to be a blocker.

Reply:
This is fine... my initial concern was to avoid nativeElement usage (and it can be done too) but it's fine. Can you create an issue and assign this to me so that we can keep track of this.